src/util/system.cpp

```diff
@@ -451,69 +358,42 @@ unsigned int ArgsManager::FlagsOfKnownArg(const std::string& key) const
 
 std::vector<std::string> ArgsManager::GetArgs(const std::string& strArg) const
 {
-    std::vector<std::string> result = {};
-    if (IsArgNegated(strArg)) return result; // special case
```

Comment:
Just to be sure the `IsArgNegated` is preserved here in `GetListSetting` via `source.span.negated() > 0` ?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/15934#discussion_r337809469

> Just to be sure the `IsArgNegated` is preserved here in `GetListSetting` via `source.span.negated() > 0` ?

It's handled implicitly by `for (const auto& value : source.span)` in `GetListSetting`, because the `SettingsSpan` iterator skips negated values in its `begin()` method.

The `source.span.negated() > 0` check you're referring is used after that to set the `result_complete` variable, which is used to apply settings precedence
`command_line_options` > `rw_settings` > `ro_settings`, and make sure for example that a negation on the command line will clear out any lower priority settings in `ro_config` and `rw_settings`. I'll add a comment to say this explicitly on the result_complete line.